File: sliding-window/121-Best-Time-to-Buy-and-Sell-Stock.go
func maxProfit(prices []int) int {
    left := 0
    ans := 0
    for right, price := range prices {
        if prices[left] > price {
            left = right
        } else {
            ans = max(ans, price - prices[left])
        }
    }
    return ans
}

func max(a int, b int) int {
    if a > b {
        return a
    }
    return b
}

File: sliding-window/121-Best-Time-to-Buy-and-Sell-Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        left = 0
        for right in range(1, len(prices)):
            if prices[left] > prices[right]:
                left = right
            else:
                currentProfit = prices[right] - prices[left]
                profit = max(profit, currentProfit)
        return profit